contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -566,13 +570,17 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
     function _sendTeleporterMessage(
         TeleporterMessageInput memory messageInput,
         TeleporterMessageReceipt[] memory receipts
-    ) private returns (uint256) {
+    ) private returns (bytes32) {
+        // If the blockchain ID has yet to be initialized, do so now.
+        bytes32 blockchainID_ = _initializeBlockchainID();
+        require(blockchainID_ != bytes32(0), "TeleporterMessenger: blockchainID not set");
+
         // Get the message ID to use for this message.
-        uint256 messageID = _getNextMessageID(messageInput.destinationBlockchainID);
+        (uint256 messageNonce_, bytes32 messageID_) = _calculateNextMessageNonceAndID();
```

Comment:
can we just use `_calculateMessageID` here? We already have the check that blockchainID != zero bytes